### DISCUSSION TITLE
Quarto crashing with tbl-colwidths

### DISCUSSION THREAD

**Discussion by fuhrmanator** [quarto-dev/quarto-cli#1274] [Q&A] [answered]
At https://quarto.org/docs/authoring/tables.html#column-widths to specify column widths in a table's caption, it's written
> `: Fruit prices {tbl-colwidths="[75,25]"}`

However, if I use this syntax in the latest Quarto 0.9.629 it crashes:

```
Error running filter C:/Users/Me/AppData/Local/Programs/Quarto/share/filters/quarto-pre/quarto-pre.lua:
...al\Programs\Quarto\bin\..\share\pandoc\datadir\_json.lua:167: bad 'for' initial value (number expected, got nil)
stack traceback:
        ...al\Programs\Quarto\bin\..\share\pandoc\datadir\_json.lua:381: in function '_json.decode'
        .../Programs/Quarto/share/filters/quarto-pre/quarto-pre.lua:2038: in upvalue 'tblColwidthValues'
        .../Programs/Quarto/share/filters/quarto-pre/quarto-pre.lua:2119: in function <.../Programs/Quarto/share/filters/quarto-pre/quarto-pre.lua:2097>
ERROR: unexpected character '�' at line 1 col 1
```

I see that it's a filter issue, and so I suspect that it's a conflict with my use of `pandoc-quotes.lua`
```yml
filters:
  - filters/pandoc-quotes.lua
  - filters/diagram-generator.lua
  - quarto
```
which turns my use of quotes into the appropriate version in the respective language, e.g. in markdown I write "Allô" and since my language is set to French, the transformation is done to « Allô » (with non-breaking spaces). I suspect that the '�' is either « or » or a non-breaking space.

Playing around with the syntax, I just dropped the double-quotes from the `tbl-colwidths="[75,25]"` i.e., `tbl-colwidths=[75,25]` and it appears to work fine.

So, I put this in the discussion since I am not sure if the problem is 1) a bug in the documentation, 2) a problem with the order of filters (should `quarto-pre.lua` go before, or is it bundled in quarto?), or 3) the use of `pandoc-quotes.lua` to accomplish the need to transform quotes (I previously was getting this by babel in LaTeX, but the pandoc way is cleaner to get to HTML and other output). 

By the way, I just wanted to say you guys rock with the bug fixes and turn-around on problems! So happy I found Quarto!

**Comment by jjallaire** [comment, marked as answer]
Thanks for reporting this! The underlying cause of this is that we don't quite have our Lua filters ordered optimally to divide into "AST Normalization" (which I'd argue the tbl-colwidths feature is) and "Features". Ideally we'd run all of our AST Normalization before user filters (so they could depend on it) but right now we only have two things in the normalization pass: (1) Merging text and file based includes; and (2) Resolving relative links within books (whose compilation occurs from the root). So for now I'd say just use the workaround you have and/or play with the order of filters (if quarto runs first then this also won't occur).
